joomla/joomla-cms
# Normalize some parameters in com_contact, com_content, com_menus xml files

Pull Request for PR #12661, that conflicted with merged PR #11911.

Normalize some parameters in com_contact, com_content, com_menus xml files (some parameters was missing in some xml, new attributes was added, see below).
Reorder some parameters as they shown in views.
Code style (https://joomla.github.io/coding-standards/?coding-standards/chapters/xml.md).

/administrator/components/com_menus/models/forms/item_component.xml - replace the option ```JGLOBAL_USE_GLOBAL``` with the attribute ```useglobal="true"```, code style.

/administrator/components/com_contact/models/forms/contact.xml - replace options ```JGLOBAL_USE_GLOBAL``` with attributes ```useglobal="true"```, code style.
/administrator/components/com_contact/config.xml - add ```show_category_heading_title_text``` parameter, reorder parameters of the ```category```, ```categories```, ```list_default_parameters``` fieldsets, code style.
/components/com_contact/models/forms/form.xml - replace options ```JGLOBAL_USE_GLOBAL``` with attributes ```useglobal="true"```, normalize language constants, code style.
/components/com_contact/views/categories/tmpl/default.xml - add ```show_category_heading_title_text```, ```show_image_heading```, ```initial_sort```, ```show_tags```, ```show_info```, ```show_user_custom_fields``` parameters, replace options ```JGLOBAL_USE_GLOBAL``` with attributes ```useglobal="true"```, add ```showon``` attribute for ```Link ... Label``` parameters, reorder parameters of the ```basic```, ```category```, ```advanced fieldsets```, code style.
/components/com_contact/views/category/tmpl/default.xml - add ```show_category_heading_title_text``` parameter, replace options ```JGLOBAL_USE_GLOBAL``` with attributes ```useglobal="true"```, add ```showon``` attribute for ```Link ... Label``` parameters, reorder parameters of the ```category``` fieldset, code style.
/components/com_contact/views/contact/tmpl/default.xml - replace options ```JGLOBAL_USE_GLOBAL``` with attributes ```useglobal="true"```, add ```showon``` attribute for ```Link ... Label``` parameters, code style.
/components/com_contact/views/featured/tmpl/default.xml - replace options ```JGLOBAL_USE_GLOBAL``` with attributes ```useglobal="true"```, add ```showon``` attribute for ```Link ... Label``` parameters, code style.

/administrator/components/com_content/config.xml - fix double names of spacers, reorder parameters of the ```category```, ```categories```, ```list_default_parameters``` fieldsets, code style.
/components/com_content/views/categories/tmpl/default.xml - add ```show_category_heading_title_text```, ```show_cat_tags```, ```page_subheading``` parameters, reorder parameters of the ```basic```, ```category```, ```advanced``` fieldsets, code style.
/components/com_content/views/category/tmpl/blog.xml - reorder parameters of the ```category``` fieldset, code style.
/components/com_content/views/category/tmpl/default.xml - remove unnecessary ```>```, reorder parameters of the ```category``` fieldset, code style.

Testing Instructions

Use a fresh Joomla install with sample items.
Create a tree of some contact categories.
Create contacts with default options and place they to some contact categories.
Create menu items of the types Contacts » List All Contact Categories and Contacts » List Contacts in a Category for the root contact category.
Check the contact categories in the frontend. All should be fine.
Play with option 'Show Subcategories Text' in Global Configuration » Contacts, in the Category tab of the menu items you created. Check everything is fine. The same for other added options.

NB: When testing List All Categories menu item other menu items must be unpublished.

The same for content (menutypes: Articles » List All Categories, Articles » Category List, Articles » Category Blog)
